lib/msal-browser/test/app/ExperimentalPublicClientApplication.spec.ts

```diff
@@ -0,0 +1,2937 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import sinon from "sinon";
+import { TEST_CONFIG, TEST_URIS, TEST_HASHES, TEST_TOKENS, TEST_DATA_CLIENT_INFO, TEST_TOKEN_LIFETIMES, RANDOM_TEST_GUID, DEFAULT_OPENID_CONFIG_RESPONSE, testNavUrl, testLogoutUrl, TEST_STATE_VALUES, testNavUrlNoRequest, DEFAULT_TENANT_DISCOVERY_RESPONSE } from "../utils/StringConstants";
+import { ServerError, Constants, AccountInfo, TokenClaims, PromptValue, AuthenticationResult, CommonAuthorizationCodeRequest, CommonAuthorizationUrlRequest, AuthToken, PersistentCacheKeys, AuthorizationCodeClient, ResponseMode, AccountEntity, ProtocolUtils, AuthenticationScheme, RefreshTokenClient, Logger, ServerTelemetryEntity, CommonSilentFlowRequest, CommonEndSessionRequest, LogLevel, NetworkResponse, ServerAuthorizationTokenResponse, CcsCredential, CcsCredentialType } from "@azure/msal-common";
+import { BrowserUtils } from "../../src/utils/BrowserUtils";
+import { BrowserConstants, TemporaryCacheKeys, ApiId, InteractionType, BrowserCacheLocation, WrapperSKU } from "../../src/utils/BrowserConstants";
+import { Base64Encode } from "../../src/encode/Base64Encode";
+import { XhrClient } from "../../src/network/XhrClient";
+import { BrowserAuthErrorMessage, BrowserAuthError } from "../../src/error/BrowserAuthError";
+import { RedirectHandler } from "../../src/interaction_handler/RedirectHandler";
+import { PopupHandler } from "../../src/interaction_handler/PopupHandler";
+import { SilentHandler } from "../../src/interaction_handler/SilentHandler";
+import { CryptoOps } from "../../src/crypto/CryptoOps";
+import { DatabaseStorage } from "../../src/cache/DatabaseStorage";
+import { EventType } from "../../src/event/EventType";
+import { SilentRequest } from "../../src/request/SilentRequest";
+import { BrowserCacheManager } from "../../src/cache/BrowserCacheManager";
+import { RedirectRequest } from "../../src/request/RedirectRequest";
+import { NavigationClient } from "../../src/navigation/NavigationClient";
+import { NavigationOptions } from "../../src/navigation/NavigationOptions";
+import { PopupUtils } from "../../src/utils/PopupUtils";
+import { EndSessionPopupRequest } from "../../src/request/EndSessionPopupRequest";
+import { EventMessage } from "../../src/event/EventMessage";
+import { EventHandler } from "../../src/event/EventHandler";
+import { ExperimentalPublicClientApplication } from "../../src/app/ExperimentalPublicClientApplication";
+import { ClientApplication } from "../../src/app/ClientApplication";
+
+describe("ExperimentalPublicClientApplication.ts Class Unit Tests", () => {
```

Comment:
Is this a carbon copy of the PCA test class?

Reply:
yes, for now

Reply:
It will probably get annoying to have to update this test class ever time something is added to PCA, so I wonder if we can find a way to ensure EPCA has the same functionality without needing to redo all the unit tests.

Reply:
+1 Can we just assert that each API exists on EPCA instead? The functionality may change anyway as we "experiment"

Reply:
Is there a way we can detect changes and test? My concern would be `Codecov` but @tnorling's suggestion probably will address that.